esx_service/utils/auth_data.py

```diff
@@ -514,15 +531,30 @@ def connect(self):
         if not self.conn:
             raise DbConnectionError(self.db_path)
         
+        # Return rows as Row instances instead of tuples
+        self.conn.row_factory = sqlite3.Row
+
         if not need_create_table:
             if os.path.isfile(self.db_path) and self.need_upgrade_db():
                 # Currently, when need upgrade, raise exception
                 # TODO: add code to handle DB schema upgrade 
                 # when schema changes
                 raise DbConnectionError(self.db_path)        
+            
+            error_msg, tenant = self.get_tenant(auth.DEFAULT_TENANT)
```

Comment:
Can we extract line 544-555 to a separate method? This piece of logic is trying to verify whether the default tenant already exists or not. It's better to keep it modulized.